src/ConstantProductFactory.sol

```diff
@@ -119,4 +206,44 @@ contract ConstantProductFactory {
         amm.token0().safeTransferFrom(msg.sender, address(amm), amount0);
         amm.token1().safeTransferFrom(msg.sender, address(amm), amount1);
     }
+
+    /**
+     * @notice Enable trading for an existing AMM that is managed by this
+     * contract.
+     * @param amm The AMM for which to enable trading.
+     * @param tradingParams The parameters used by the CoW AMM to create the
+     * order.
+     */
+    function _enableTrading(ConstantProduct amm, ConstantProduct.TradingParams memory tradingParams) internal {
+        amm.enableTrading(tradingParams);
+        emit TradingEnabled(amm, msg.sender);
+        // The salt is unused by this contract. However, external tools (for
+        // example the watch tower) may expect that the salt doesn't repeat.
+        // This cannot be achieved without extra state or bubbling up the nonce
+        // parameter to the caller.
+        // We assume that external tools are able to handle the case where the
+        // exact same order is created two times in the same block for the
+        // exact same parameters, AMM address, and owner.
+        // Note that there can be at most one active order per AMM by
+        // construction, and a repeating salt means that the order was created,
+        // cancelled, and then recreated with the same parameters on the same
+        // block.
+        bytes32 salt = bytes32(bytes20(msg.sender)) | bytes32(block.timestamp);
+        // The following event will be pickd up by the watchtower offchain
+        // service, which is responsible for automatically posting CoW AMM
+        // orders on the CoW Protocol orderbook.
+        emit ComposableCoW.ConditionalOrderCreated(
+            address(amm),
+            IConditionalOrder.ConditionalOrderParams(IConditionalOrder(address(this)), salt, abi.encode(tradingParams))
+        );
```

Comment:
I think we can simply this and use `bytes32(0)` for the salt. The only expectation that is enforced somewhat for off-chain is that `keccak(IConditionalOrder.ConditionalOrderParams)` per `owner` (ie. `amm` in this case) is unique.

Reply:
> The only expectation [...] is that `keccak(IConditionalOrder.ConditionalOrderParams)` per owner [...] is unique.

Isn't then the salt necessary? It seems reasonable to me that a user could change parameters and then go back to the same parameters for the same AMM. As long as this doesn't happen in the same block, the salt as defined here preserves this expectation.

Reply:
While this may be the case, the watch-tower will still pick it up again. The original conditional order may or may not have been purged, but nonetheless, the most recent instance will be picked up and executed.

Reply:
Set salt to zero in 3b57578ad27007088d09b79e60416becf59c3b6a.